src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -368,6 +396,30 @@ public async Task<RestoreResult> ExecuteAsync(CancellationToken token)
             }
         }
 
+        private async Task<bool> AreCentralVersionRequirementsSatisfiedAsync(bool cpvmEnabled)
```

Comment:
point of discussion, really not a request for change: This API feels a bit weird to me. The caller knows going in that CPVM is not enabled, so why call the method? This method is only called in one place, so it can't be to reduce duplicate code, but even that would be a difficult position to defend. Call caller could just change the if statement to `if (cpvmEnabled && await AreCentalVersionRequrementsSatisfiedAsync())`.